### PR TITLE
fix a bug that Collection from backlink holds wrong table pointer.

### LINKED ISSUE #4487: RealmObjects from linking objects hold wrong table.
Following test fails because of `Table` object in `sourceFromBacklinks` is wrong.

```
        realm.beginTransaction();
        final BacklinksTarget target = realm.createObject(BacklinksTarget.class);
        target.setId(1);
        final BacklinksSource source = realm.createObject(BacklinksSource.class);
        source.setChild(target);
        realm.commitTransaction();

        final RealmResults<BacklinksSource> parents = target.getParents();
        final BacklinksSource sourceFromBacklinks = parents.first();

        assertEquals(source, sourceFromBacklinks);
```

The `Table` object is only used in `equals()`, `hashCode()`, and default value assignment.
Since RealmObjects from linking objects never set default values, this bug affects only `equals()` and `hashCode()`.

### PR DESCRIPTION
**Pull request by zaki50**
fixes #4487 

@realm/java

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java [hunk 1]
```diff
@@ -151,6 +151,22 @@ public void basic_multipleReferencesFromParentList() {
         assertEquals(parent, child.getListParents().last());
     }
 
+    // This test reproduces https://github.com/realm/realm-java/issues/4487
+    @Test
+    public void issue4487() {
```
Comment (beeender) [inline realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java]: something like `issue4487_checkIfTableIsCorrect`?

Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java]: done

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+## 3.1.3 (YYYY-MM-DD)
+
+### Bug Fixes
+
+* `equals()` and `hashCode()` of managed `RealmObject`s that come from linking objects don't work correctly (#4487).
+
+
```
Comment (dalinaum) [inline CHANGELOG.md]: How about removing an additional new line to keep consistent?

Comment (zaki50) [inline CHANGELOG.md]: done

**Pull request merged**